### PR TITLE
just: add module for the just command runner

### PR DESCRIPTION
**Pull request by maximsmol**
### Description

Adds a module for [just](https://github.com/casey/just) and its shell completions.

### Checklist

- [X] Change is backwards compatible.

- [X] Code formatted with `./format`.

- [X] Code tested through `nix-shell --pure tests -A run.all`.

- [X] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [X] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [X] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [X] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -437,3 +437,5 @@ Makefile                                              @thiagokokada
 /modules/programs/ion.nix                             @jo1gi
 
 /modules/services/plex-mpv-shim.nix                   @starcraft66
+
+/modules/programs/just.nix                            @maximsmol
```
Comment (berbiche) [inline .github/CODEOWNERS:441]: Could this be moved at the right place to preserve the ordering? Thanks.

File: modules/programs/just.nix [hunk 2]
```diff
@@ -0,0 +1,64 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.just;
+
+in {
+  meta.maintainers = [ maintainers.maximsmol ];
```
Comment (berbiche) [inline modules/programs/just.nix:10]:
```suggestion
  meta.maintainers = [ hm.maintainers.maximsmol ];
```

Home Manager's maintainers are available `lib.hm.maintainers`.

File: modules/programs/just.nix [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  options.programs.just = {
+    enable = mkEnableOption
+      "just is a handy way to save and run project-specific commands";
```
Comment (berbiche) [inline modules/programs/just.nix:14]: This expands to `Whether to enable just is a handy way...`.
The phrasing seems odd.

**Review by berbiche — CHANGES_REQUESTED** [review]
Thanks for the contribution!

File: modules/programs/just.nix [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+  cfg = config.programs.just;
+
+in
+{
+  meta.maintainers = [ hm.maintainers.maximsmol ];
+
+  options.programs.just = {
+    enable = mkEnableOption
+      "just - a handy way to save and run project-specific commands";
```
Comment (sumnerevans) [inline modules/programs/just.nix:15]:
```suggestion
      "just, a handy way to save and run project-specific commands";
```

File: modules/programs/just.nix [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      description = "Package providing the <command>just</command> tool.";
+    };
+
+    enableBashIntegration = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to enable Bash integration.
+      '';
+    };
```
Comment (sumnerevans) [inline modules/programs/just.nix:30]: All of these enable options can use `mkEnableOption` like so:

```suggestion
    enableBashIntegration = mkEnableOption "Bash integration" // {
      default = true;
    };
```

**Comment by maximsmol** [comment]
@sumnerevans just fyi, copied both the description style ("{name} - {description}") and the `mkOption` from `programs.fzf`—I could reformat those in another PR according to your suggestions as well if you think it'd be good

**Review by berbiche — APPROVED** [review]
LGTM!

**Comment by berbiche** [comment]
Your PR needs to be rebased on "master" to include a fix that will allow the checks to succeed.

**Comment by rycee** [comment]
Merged to master now. Thanks!

**Pull request merged**